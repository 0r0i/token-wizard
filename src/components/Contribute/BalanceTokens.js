import React from 'react'

export const BalanceTokens = ({ balance, ticker, extraClassName = '' }) => (
  <div className={`ba-BalanceTokens ${extraClassName}`}>
    <h2 className="ba-BalanceTokens_Title">
      {balance} <span className="ba-BalanceTokens_Title-uppercase">{ticker}</span>
    </h2>
    <p className="ba-BalanceTokens_Description">Your Balance</p>
  </div>
)
