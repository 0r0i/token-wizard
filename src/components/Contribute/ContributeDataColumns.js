import React from 'react'

export const ContributeDataColumns = ({ extraClassName = '', data }) => (
  <div className={`cnt-ContributeDataColumns ${extraClassName}`}>
    {data.map((item, index) => (
      <div className="cnt-ContributeDataColumns_Item" key={index}>
        <p className="cnt-ContributeDataColumns_ItemData">{item.title}</p>
        <h4 className="cnt-ContributeDataColumns_ItemTitle">{item.description}</h4>
      </div>
    ))}
  </div>
)
