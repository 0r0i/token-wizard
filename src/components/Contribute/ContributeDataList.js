import React from 'react'
import { ButtonCopyToClipboard } from '../Common/ButtonCopyToClipboard'

export const ContributeDataList = ({ currentAccount, crowdsaleAddress, extraClassName = '' }) => (
  <div className={`cnt-ContributeDataList ${extraClassName}`}>
    {currentAccount ? (
      <div className="cnt-ContributeDataList_Item">
        <p className="cnt-ContributeDataList_ItemData">
          <span className="cnt-ContributeDataList_ItemDataValue">{currentAccount}</span>
          <ButtonCopyToClipboard value={currentAccount} />
        </p>
        <h4 className="cnt-ContributeDataList_ItemTitle">Current Account</h4>
      </div>
    ) : null}
    {crowdsaleAddress ? (
      <div className="cnt-ContributeDataList_Item">
        <p className="cnt-ContributeDataList_ItemData">
          <span className="cnt-ContributeDataList_ItemDataValue">{crowdsaleAddress}</span>{' '}
          <ButtonCopyToClipboard value={crowdsaleAddress} />
        </p>
        <h4 className="cnt-ContributeDataList_ItemTitle">Proxy Address</h4>
      </div>
    ) : null}
  </div>
)
