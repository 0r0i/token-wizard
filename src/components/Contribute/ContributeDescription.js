import React from 'react'

export const ContributeDescription = ({ extraClassName = '' }) => (
  <div className={`cnt-ContributeDescription ${extraClassName}`}>
    Here you can contribute in the crowdsale campaign. At the moment, you need{' '}
    <a
      target="_blank"
      rel="noopener noreferrer"
      href="https://chrome.google.com/webstore/detail/nifty-wallet/jbdaocneiiinmjbjlgalhcelgbejmnid"
    >
      a Wallet
    </a>{' '}
    client to contribute into the crowdsale.
  </div>
)
