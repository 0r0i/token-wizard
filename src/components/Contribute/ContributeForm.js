import React from 'react'
import { inject, observer } from 'mobx-react'
import { Field, FormSpy } from 'react-final-form'
import { CONTRIBUTION_OPTIONS } from '../../utils/constants'
import { InputField2 } from '../Common/InputField2'
import {
  composeValidators,
  isDecimalPlacesNotGreaterThan,
  isGreaterOrEqualThan,
  isRequired
} from '../../utils/validations'
import { TokenDimension } from '../Common/TokenDimension'

export const ContributeForm = inject('contributeStore', 'tokenStore')(
  observer(({ handleSubmit, pristine, invalid, ...props }) => {
    const {
      contributeThrough,
      updateContributeThrough,
      web3Available,
      contributeStore,
      tokenStore,
      isEnded,
      isFinalized,
      isStarted,
      isSoldOut,
      isTierSoldOut
    } = props
    const { decimals } = tokenStore
    const buttonDisabled = pristine || invalid || isEnded || isFinalized || isSoldOut || isTierSoldOut

    const contributeThroughWallet = () => {
      return contributeThrough === CONTRIBUTION_OPTIONS.METAMASK
    }

    const ContributeButton = contributeThroughWallet() ? (
      <button className="cnt-ContributeForm_ContributeButton" onClick={handleSubmit} disabled={buttonDisabled}>
        Contribute
      </button>
    ) : null

    const canContribute = !(isEnded || isFinalized || isSoldOut)

    const validateContribute = value => {
      const decimalsErr = `Number of tokens to buy should be positive and should not exceed ${decimals} decimals.`
      const minimumContributionErr =
        isFinite(props.minimumContribution) && canContribute
          ? `Minimum valid contribution: ${props.minimumContribution}`
          : `You are not allowed`
      let errors = composeValidators(
        isRequired(),
        isDecimalPlacesNotGreaterThan(decimalsErr)(decimals),
        isGreaterOrEqualThan(minimumContributionErr)(props.minimumContribution)
      )(value)

      if (!isStarted && errors && errors.length > 0) {
        errors = ['You are not allowed']
      }

      if (errors) return errors.shift()
    }

    const tokensToContributeOnChange = ({ values }) => {
      if (values && values.contribute !== undefined) {
        contributeStore.setProperty('tokensToContribute', values.contribute)
      }
    }

    return (
      <form className={`cnt-ContributeForm`} onSubmit={handleSubmit}>
        <h3 className="cnt-ContributeForm_Title">Choose amount to contribute</h3>
        <div className="cnt-ContributeForm_AmountContainer">
          <Field
            component={InputField2}
            extraClassName="cnt-ContributeForm_Amount"
            name="contribute"
            placeholder="0"
            validate={validateContribute}
          />
          <TokenDimension type="tokens" />
        </div>
        <select value={contributeThrough} className="sw_Select" onChange={e => updateContributeThrough(e.target.value)}>
          <option disabled={!web3Available} value={CONTRIBUTION_OPTIONS.METAMASK}>
            Wallet {!web3Available ? ' (not available)' : null}
          </option>
          <option value={CONTRIBUTION_OPTIONS.QR}>QR</option>
        </select>
        {ContributeButton}
        <FormSpy subscription={{ values: true }} onChange={tokensToContributeOnChange} />
      </form>
    )
  })
)
