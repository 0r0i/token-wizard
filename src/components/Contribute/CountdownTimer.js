import React from 'react'
import Circle from 'react-circle'
import { pad } from '../../utils/utils'
import logdown from 'logdown'

const logger = logdown('TW:CountDownTimer')

const CountdownTimer = ({
  altMessage,
  crowdsaleTimePassedPercentage,
  days,
  hours,
  isFinalized,
  isSoldOut,
  isTierSoldOut,
  isLoading,
  minutes,
  nextTick,
  seconds,
  tiersLength
}) => {
  const noTimeLeft = days <= 0 && hours <= 0 && minutes <= 0 && seconds <= 0
  const crowdsaleIsRunning = !isFinalized && nextTick && nextTick.type
  const hoursHaveBlinker = days ? 'cnt-CountdownTimer_TimeItem-blinking' : ''

  const countdownClock = (
    <div className="cnt-CountdownTimer_Time">
      {days ? (
        <div className="cnt-CountdownTimer_TimeItem">
          <div className="cnt-CountdownTimer_TimeCount">{days}</div>
          <div className="cnt-CountdownTimer_TimeInterval">Days</div>
        </div>
      ) : null}
      <div className={`cnt-CountdownTimer_TimeItem ${hoursHaveBlinker}`}>
        <div className="cnt-CountdownTimer_TimeCount">{pad(hours, 2)}</div>
        <div className="cnt-CountdownTimer_TimeInterval">Hours</div>
      </div>
      <div className="cnt-CountdownTimer_TimeItem">
        <div className="cnt-CountdownTimer_TimeCount">{pad(minutes, 2)}</div>
        <div className="cnt-CountdownTimer_TimeInterval">Mins</div>
      </div>
      {days ? null : (
        <div className="cnt-CountdownTimer_TimeItem">
          <div className="cnt-CountdownTimer_TimeCount">{pad(seconds, 2)}</div>
          <div className="cnt-CountdownTimer_TimeInterval">Secs</div>
        </div>
      )}
    </div>
  )

  const getCrowdsaleMessage = () => {
    let theMessage

    logger.log('Is Finalized', isFinalized)
    logger.log('Is sold out', isSoldOut)
    logger.log('Is Tier sold out', isTierSoldOut)
    if (isFinalized) {
      theMessage = 'crowdsale has been finalized'
    } else {
      if (nextTick && nextTick.type) {
        if (nextTick.type === 'start' && !noTimeLeft) {
          theMessage = `To start of tier ${nextTick.order} of ${tiersLength}`
        } else if (nextTick.type === 'end' && !noTimeLeft && isSoldOut && nextTick.order === tiersLength) {
          theMessage = 'Crowdsale has ended'
        } else if (nextTick.type === 'end' && !noTimeLeft) {
          theMessage = `To end of tier ${nextTick.order} of ${tiersLength}`
        } else {
          theMessage = `Waiting...`
        }
      } else {
        theMessage = 'Crowdsale has ended'
      }
    }

    return <div className="cnt-CountdownTimer_Message">{theMessage}</div>
  }

  const getCrowdsaleAltMessage = () => {
    return altMessage ? <div className="cnt-CountdownTimer_AltMessage">{altMessage}</div> : null
  }

  const getCrowdsaleProgress = () => {
    if (noTimeLeft || !crowdsaleIsRunning || isFinalized || isSoldOut) {
      return 100
    }

    return crowdsaleTimePassedPercentage
  }

  return (
    <div className="cnt-CountdownTimer">
      <div className="cnt-CountdownTimer_ReactCountdownClock">
        <Circle
          animate={true}
          bgColor="#5a2da5"
          lineWidth="50"
          progress={getCrowdsaleProgress()}
          progressColor="#a971f9"
          responsive={false}
          roundedStroke={false}
          showPercentage={false}
          size="250"
        />
        {isLoading ? null : (
          <div className="cnt-CountdownTimer_TimeContainer">
            {noTimeLeft || isFinalized || isSoldOut ? null : countdownClock}
            {getCrowdsaleMessage()}
            {getCrowdsaleAltMessage()}
          </div>
        )}
      </div>
    </div>
  )
}

export default CountdownTimer
