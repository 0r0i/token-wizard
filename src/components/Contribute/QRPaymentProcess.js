import React from 'react'
import QRCode from 'qrcode.react'
import { ButtonCopyToClipboard } from '../Common/ButtonCopyToClipboard'

const QRPaymentProcess = ({ crowdsaleProxyAddr, registryExecAddr, txData }) => {
  const targetContractStr = crowdsaleProxyAddr ? 'proxy' : 'RegistryExec'
  const targetAddr = crowdsaleProxyAddr ? crowdsaleProxyAddr : registryExecAddr

  return (
    <div className="cnt-QRPaymentProcess">
      <div className="cnt-QRPaymentProcess_QR">
        <QRCode fgColor="#5a2da5" value={targetAddr} size={115} bgColor="#f6f6f6" />
      </div>
      <div className="cnt-QRPaymentProcess_HashContainer">
        <span className="cnt-QRPaymentProcess_Hash">{targetAddr}</span>
        <ButtonCopyToClipboard value={targetAddr} />
      </div>
      <div className="cnt-QRPaymentProcess_Note">
        <h3 className="cnt-QRPaymentProcess_NoteTitle">Important</h3>
        <p className="cnt-QRPaymentProcess_NoteDescription">
          Send ethers to the crowdsale {targetContractStr} address with a data:{' '}
        </p>
        <div className="cnt-QRPaymentProcess_txDataContainer">
          <span className="cnt-QRPaymentProcess_txData">{txData}</span>
          <ButtonCopyToClipboard buttonColor="#270167" value={txData} />
        </div>
      </div>
    </div>
  )
}

export default QRPaymentProcess
