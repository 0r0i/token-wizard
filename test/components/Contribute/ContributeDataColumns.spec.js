import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'
import { ContributeDataColumns } from '../../../src/components/Contribute/ContributeDataColumns'

configure({ adapter: new Adapter() })

const data = [
  {
    description: 'Token Name',
    title: 'Name'
  },
  {
    description: 'TKN',
    title: 'Ticker'
  },
  {
    description: '100 TKN',
    title: 'Total Supply'
  },
  {
    description: 10,
    title: 'Minimum Contribution'
  },
  {
    description: 100,
    title: 'Maximum Contribution'
  }
]

describe(`ContributeDataColumns`, () => {
  it(`should contain a data.length amount of '.cnt-ContributeDataColumns_Item' items`, () => {
    const wrapper = mount(<ContributeDataColumns data={data} />)
    const dataItems = wrapper.find('.cnt-ContributeDataColumns_Item')

    expect(dataItems.length).toBe(data.length)
  })

  it(`should render ContributeDataColumns component`, () => {
    // Given
    const component = render.create(<ContributeDataColumns data={data} />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
