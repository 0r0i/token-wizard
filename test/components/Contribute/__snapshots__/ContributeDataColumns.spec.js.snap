// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributeDataColumns should render ContributeDataColumns component 1`] = `
<div
  className="cnt-ContributeDataColumns "
>
  <div
    className="cnt-ContributeDataColumns_Item"
  >
    <p
      className="cnt-ContributeDataColumns_ItemData"
    >
      Name
    </p>
    <h4
      className="cnt-ContributeDataColumns_ItemTitle"
    >
      Token Name
    </h4>
  </div>
  <div
    className="cnt-ContributeDataColumns_Item"
  >
    <p
      className="cnt-ContributeDataColumns_ItemData"
    >
      Ticker
    </p>
    <h4
      className="cnt-ContributeDataColumns_ItemTitle"
    >
      TKN
    </h4>
  </div>
  <div
    className="cnt-ContributeDataColumns_Item"
  >
    <p
      className="cnt-ContributeDataColumns_ItemData"
    >
      Total Supply
    </p>
    <h4
      className="cnt-ContributeDataColumns_ItemTitle"
    >
      100 TKN
    </h4>
  </div>
  <div
    className="cnt-ContributeDataColumns_Item"
  >
    <p
      className="cnt-ContributeDataColumns_ItemData"
    >
      Minimum Contribution
    </p>
    <h4
      className="cnt-ContributeDataColumns_ItemTitle"
    >
      10
    </h4>
  </div>
  <div
    className="cnt-ContributeDataColumns_Item"
  >
    <p
      className="cnt-ContributeDataColumns_ItemData"
    >
      Maximum Contribution
    </p>
    <h4
      className="cnt-ContributeDataColumns_ItemTitle"
    >
      100
    </h4>
  </div>
</div>
`;
